client/src/pages/admin/Index.js

```diff
@@ -204,9 +204,9 @@ const AdminIndex = ({ pageDispatchers }) => {
   }
 
   function onSubmitSuccess(response, values, form) {
-    // After successful submit, reset the form in order to make sure the dirty
-    // prop is also reset (otherwise we would get a blocking navigation warning)
-    form.resetForm()
+    // reset the form to latest values
```

Comment:
Same goes for all places you added the comment.

Reply:
I don't remember saying this was because of unnecessary renders, but anyway prev version was the cause of "unsaved changes" prompt, since the report is initialized to a starting state, invisible fields etc needs to be set after that, which makes the values dirty and cause unnecessary renders as well.